Timeline.cs

```diff
@@ -0,0 +1,507 @@
+//By Lorenzo Lotti
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// A collection of <see cref="System.DateTime"/> and <see cref="TValue"/> sorted by <see cref="System.DateTime"/> field.
+/// </summary>
+/// <typeparam name="TValue">Value associated with a <see cref="System.DateTime"/>.</typeparam>
+internal class Timeline<TValue>:
+    ICollection<(DateTime Time, TValue Value)>,
+    IEquatable<Timeline<TValue>>,
+    ICloneable
+{
+    private List<(DateTime Time, TValue Value)> timeline;
+    bool ICollection<(DateTime Time, TValue Value)>.IsReadOnly { get => false; }
+
+    /// <summary>
+    /// The count of items.
+    /// </summary>
+    public int Count { get => timeline.Count; }
+
+    /// <summary>
+    /// Get all values associated with <paramref name="time"/>.
+    /// </summary>
+    /// <returns>Values associated with <paramref name="time"/>.</returns>
+    public TValue[] this[DateTime time]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use string or an integral type here, or refactor this indexer into a method.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5574777)